python/triton/language/semantic.py

```diff
@@ -1133,6 +1133,10 @@ def reduce_impl(input: tl.tensor, axis: int, builder: ir.builder, name: str,
 
     # get result type
     shape = input.type.shape
+
+    rank = len(shape)
+    assert axis >= 0 and axis < rank, f"axis (v={axis}) is out of range, should be within [0, {rank})"
```

Comment:
The condition could be simplified to `0 <= axis < rank`.

Reply:
Pythonic, Thanks